src/berry_mill/builder.py

```diff
@@ -0,0 +1,198 @@
+
+        if machine() == "aarch64":
+            box_options += ["--machine", "virt"]
+
+        allow_no_accel:bool = True
+        # TODO: When using cross, e.g. cpu param needs to be disabled
+        if self._params.get("cross") and machine() == "x86_64":
+            box_options += ["--aarch64", "--cpu", "cortex-a57", "--machine", "virt", "--no-accel"]
+            kiwi_options += ["--target-arch", "aarch64"]
+            allow_no_accel = False
+
+        if self._params.get("accel", False) and allow_no_accel:
+            box_options.append("--no-accel")
+
+        try:
+            config_tree = etree.parse(f"{self._appliance_descr}")
+        except Exception as err:
+            print("ERROR: Failure {} while parsing appliance description".format(err))
+            return
+
+        try:
+            image_name:list = config_tree.xpath("//image/@name")
+        except Exception as err:
+            print("ERROR: Failure {} while trying to extract image name", err)
+            return
+
+        target_dir = os.path.join(
+            self._params.get("target_dir","/tmp"),\
+            f"{image_name[0]}.{self._kiwiparams.get('profile', '')}")
+
+        if self._params.get("clean", False):
+            clean_target = target_dir
+            if self._params.get("profile"):
+                clean_target += f".{self._params.get('profile')}"
+            if self._params.get("local", False):
+                shutil.rmtree(clean_target)
+
+        if not self._params.get("local", False):
+            self._write_repokeys_box(self._repos)
+
+        if self._params.get("local", False):
+            command = ["kiwi-ng"] + kiwi_options\
+                    + ["system", "build", "--description", "."]\
+                    + ["--target-dir", target_dir] 
+
+            try:
+                print("Starting Kiwi for local build")                
+                KiwiAppLocal(command, repos=self._repos).run()
+            except KiwiError as kiwierr:
+                print("KiwiError:", type(kiwierr).__name__)
+                print(kiwierr)
+                return
+        else:
+            command = ["kiwi-ng"]+ kiwi_options\
+                    + ["system", "boxbuild"] + box_options\
+                    + ["--", "--description", '.'] + ["--target-dir", target_dir]\
+
+            try:
+                print("Starting Kiwi Box")
+                KiwiAppBox(command, repos=self._repos, args_tmp_dir=self._boxtmpargdir).run()
+            except KiwiError as kiwierr:
+                print("KiwiError:", type(kiwierr).__name__)
+                print(kiwierr)
+                return
+
+        # run kiwi here with "appliance_init" which is a ".kiwi" file
+        os.system("ls -lah")
+
+        print(self._repos)
+
+
+
+    def cleanup(self) -> None:
+        """
+        Cleanup the environment after the build
+        """
+        super().cleanup()
+        if not self._initialized:
+            print("Cleanup finished")
```

Comment:
This is probably a debug level of a logging facility instead.

Reply:
done